File: 1_2_MinMax.cpp
#include <iostream>
#include <algorithm>
using namespace std;

class MinMax
{
	int arr[100], n;
	
public:
	
	void getData()
	{
		cout<<"Enter number of elements: ";
		cin>>n;
		
		cout<<"Enter elements:\n";
		
		for(int i=0;i<n;i++){
			cin>>arr[i];
		}
	}
	
	void findMinMax(int low, int high, int &mini, int &maxi){
		int mid;
		int min1, max1;
		int min2, max2;
		
		if(low == high)
		{
			mini = maxi = arr[low];
		}
		
		else if(high==low+1)
		
		{

			mini = min(arr[low],arr[high]);
			maxi = max(arr[high],arr[low]);

	}
	
	else{
		mid = (low + high)/2;
		
		findMinMax(low,mid,min1,max1);
		
		findMinMax(mid+1,high,min2,max2);
		
	
		mini = min(min1,min2);
		maxi = max(max1,max2);

	}
}
	
	void display(){
		int mini,maxi;
		
		findMinMax(0,n-1,mini,maxi);
		
		cout<<"\nMinimum element = "<<mini;
		cout<<"\nMaximum element = "<<maxi;
	}
};

int main(){
	MinMax m;
	m.getData();
	m.display();
	
	return 0;
}

File: 1_2_MinMax.java
import java.util.Scanner;

public class MinMax {
    private int[] arr = new int[100];
    private int n;
    private Scanner sc = new Scanner(System.in);

    public void getData() {
        System.out.print("Enter number of elements: ");
        n = sc.nextInt();

        System.out.println("Enter elements:");
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
    }

    public int[] findMinMax(int low, int high) {
        int[] result = new int[2]; // [min, max]
        int mid;
        int[] min1max1;
        int[] min2max2;

        if (low == high) {
            result[0] = arr[low];
            result[1] = arr[low];
        } else if (high == low + 1) {
            result[0] = Math.min(arr[low], arr[high]);
            result[1] = Math.max(arr[high], arr[low]);
        } else {
            mid = (low + high) / 2;

            min1max1 = findMinMax(low, mid);
            min2max2 = findMinMax(mid + 1, high);

            result[0] = Math.min(min1max1[0], min2max2[0]);
            result[1] = Math.max(min1max1[1], min2max2[1]);
        }

        return result;
    }

    public void display() {
        int[] result = findMinMax(0, n - 1);
        System.out.println("\nMinimum element = " + result[0]);
        System.out.println("Maximum element = " + result[1]);
    }

    public static void main(String[] args) {
        MinMax m = new MinMax();
        m.getData();
        m.display();
    }
}
